base/number.jl

```diff
@@ -233,7 +233,8 @@ julia> widemul(Float32(3.), 4.)
 """
 widemul(x::Number, y::Number) = widen(x)*widen(y)
 
-iterate(x::Number, done = false) = done ? nothing : (x, true)
+iterate(x::Number) = (x, true)
```

Comment:
Should never matter (after inference, inlining, or SROA), but constructing `(x, nothing)` would require 1 byte less space. Maybe use that instead?

Reply:
Yes. I've actually changed that locally to see whether it makes a difference with the failure in `make julia-debug` (which it doesn't), and realized that it might be better anyway.